src/clients/ynab-client.ts

```diff
@@ -0,0 +1,75 @@
+import * as ynab from "ynab";
+import * as statusAnd from "../utils/statusAnd";
+import {DateTime} from "luxon"
+
+export class YnabClient {
+  private accessToken: string;
+  private client: ynab.API
+
+  constructor(token: string) {
+    this.accessToken = token;
+    this.client = new ynab.API(token);
+  }
+
+  async getBudgets() {
+    const response = await this.client.budgets.getBudgets();
+    const budgets = response.data.budgets;
+
+    if (!budgets.length) {
+      throw statusAnd.error('API call to get budgets was successful, but returned 0 budgets!');
```

Comment:
I haven't come across this setup in node projects before, felt it was a bit confusing tbh.  up to you!

Reply:
I am open to other ideas! The approach you're taking is to throw an exception, but I would say that exceptions should be used only for truly exceptional cases, and then we still have the need for cases that we need to surface a warning to the user because we couldn't complete the action.